### PR TITLE
[2/3] Update `.pyi` Python stub files: Prettify `rnn.py` by using type annotated `NamedTuple`

### PR DESCRIPTION
**Pull request by XuehaiPan**
Changes:

- #95200

1. Recognize `.py.in` and `.pyi.in` files as Python in VS Code for a better development experience.
2. Fix deep setting merge in `tools/vscode_settings.py`.

- => this PR: #95267

3. Use `Namedtuple` rather than `namedtuple + __annotations__` for `torch.nn.utils.rnn.PackedSequence_`:

    `namedtuple + __annotations__`:

    ```python
    PackedSequence_ = namedtuple('PackedSequence_',
                                 ['data', 'batch_sizes', 'sorted_indices', 'unsorted_indices'])

    # type annotation for PackedSequence_ to make it compatible with TorchScript
    PackedSequence_.__annotations__ = {'data': torch.Tensor, 'batch_sizes': torch.Tensor,
                                       'sorted_indices': Optional[torch.Tensor],
                                       'unsorted_indices': Optional[torch.Tensor]}
    ```

    `Namedtuple`: Python 3.6+

    ```python
    class PackedSequence_(NamedTuple):
        data: torch.Tensor
        batch_sizes: torch.Tensor
        sorted_indices: Optional[torch.Tensor]
        unsorted_indices: Optional[torch.Tensor]
    ```

- #95268

4. Sort import statements and remove unnecessary imports in `.pyi`, `.pyi.in` files.
5. Format `.pyi`, `.pyi.in` files and remove unnecessary ellipsis `...` in type stubs.

### CODE REVIEW

File: torch/ao/nn/quantized/dynamic/modules/rnn.py [hunk 1]
```diff
@@ -456,10 +456,10 @@ def forward_packed(
     ) -> Tuple[PackedSequence, Tuple[Tensor, Tensor]]:
         input_, batch_sizes, sorted_indices, unsorted_indices = input
         max_batch_size = batch_sizes[0]
```
Comment (janeyx99) [inline torch/ao/nn/quantized/dynamic/modules/rnn.py:458]:
```suggestion
        max_batch_size = int(batch_sizes[0])
```
Why not this?

File: torch/nn/utils/rnn.py [hunk 2]
```diff
@@ -1,24 +1,21 @@
...
            'unpad_sequence', 'pack_sequence', 'unpack_sequence']
 
-PackedSequence_ = namedtuple('PackedSequence_',
-                             ['data', 'batch_sizes', 'sorted_indices', 'unsorted_indices'])
 
-# type annotation for PackedSequence_ to make it compatible with TorchScript
-PackedSequence_.__annotations__ = {'data': torch.Tensor, 'batch_sizes': torch.Tensor,
-                                   'sorted_indices': Optional[torch.Tensor],
-                                   'unsorted_indices': Optional[torch.Tensor]}
+class PackedSequence_(NamedTuple):
```
Comment (janeyx99) [inline torch/nn/utils/rnn.py:14]: @davidberard98 Is there anything we need to verify here to ensure TorchScript doesn't break? (or is CI sufficient?)

**Review by janeyx99 — COMMENTED** [review]
Could you update the PR title to explain what this does? Something like
"Prettify rnn.py by using type annotated NamedTuple"

File: torch/ao/nn/quantized/dynamic/modules/rnn.py [hunk 3]
```diff
@@ -456,10 +456,10 @@ def forward_packed(
...
-        max_batch_size = int(max_batch_size)
 
         output_, hidden = self.forward_impl(
-            input_, hx, batch_sizes, max_batch_size, sorted_indices)
+            input_, hx, batch_sizes, int(max_batch_size), sorted_indices
```
Comment (XuehaiPan) [inline torch/ao/nn/quantized/dynamic/modules/rnn.py:461]:
```suggestion
            input_, hx, batch_sizes, max_batch_size, sorted_indices
```

File: torch/ao/nn/quantized/dynamic/modules/rnn.py [hunk 4]
```diff
@@ -702,9 +702,9 @@ def forward_packed(
     ) -> Tuple[PackedSequence, Tensor]:
         input_, batch_sizes, sorted_indices, unsorted_indices = input
         max_batch_size = batch_sizes[0]
```
Comment (XuehaiPan) [inline torch/ao/nn/quantized/dynamic/modules/rnn.py:704]:
```suggestion
        max_batch_size = int(batch_sizes[0])
```

File: torch/nn/utils/rnn.py (hunk 2 above)
Comment (XuehaiPan) [reply torch/nn/utils/rnn.py:14]: Inheriting from the typed `NamedTuple` is doing exact same thing as the old approach:

```python
In [1]: from typing import *

In [2]: import torch

In [3]: class PackedSequence_(NamedTuple):
   ...:     data: torch.Tensor
   ...:     batch_sizes: torch.Tensor
   ...:     sorted_indices: Optional[torch.Tensor]
   ...:     unsorted_indices: Optional[torch.Tensor]
   ...:     

In [4]: PackedSequence_.__annotations__
Out[4]: {'data': <class 'torch.Tensor'>, 'batch_sizes': <class 'torch.Tensor'>, 'sorted_indices': typing.Optional[torch.Tensor], 'unsorted_indices': typing.Optional[torch.Tensor]}
```

Comment (davidberard98) [reply torch/nn/utils/rnn.py:14]: looks fine for me re: torchscript (at least the changes in this file with NamedTuple)

**Review by janeyx99 — APPROVED** [review]

**Comment by janeyx99** [comment]
Feel free to land this PR whenever you are ready

**Pull request closed without merging**